### DISCUSSION TITLE
Strange final check in relocalizer.

### DISCUSSION THREAD

**Discussion by MichaelScharrer** [stella-cv/stella_vslam#420] [Q&A] [answered]
I'm having trouble understanding the purpose of following line:
https://github.com/stella-cv/stella_vslam/blob/main/src/stella_vslam/module/relocalizer.cc#L322
As far as I understand, this rejects the relocalization if there are more than 0.8 times as many optimization inliers as tracked landmarks in the candidate keyframe. However, the optimization also uses landmarks from neighboring keyframes. I would therefore expect a good relocalization to have more optimization inliers, and this is indeed the case for me.
Am I missing something?

**Comment by ymd-stella** [comment, marked as answer]
The logic seems to be reversed. This is a recently implemented feature, so there may be mistakes. I will research it.

**Reply by ymd-stella** [reply]
https://github.com/stella-cv/stella_vslam/pull/421

**Reply by MichaelScharrer** [reply]
Thanks, the new version makes a lot more sense.
